### ISSUE TITLE
CIImageProcessor not applied to downloaded image

### ISSUE THREAD

**Issue by vzsg** [onevcat/Kingfisher#485]
I'm trying to follow the "Create a processor from CIFilter" example from the cheat sheet with Kingfisher 3.1.3, but the filter is not getting called.

My processor class is [here](https://gist.github.com/vzsg/964a307575fc2c534e90bd294b93ed8d). And here's how I try to use it:

```swift
let url: NSURL = ...
imageView.kf.setImage(with: url,
                           placeholder: nil,
                           options: [.processor(WhiteToTransparentProcessor())])
```

The filter function is never called, the image appears in its original form. The `.forceRefresh` option has no effect on it, so it's not caching related. (The filter works elsewhere.)

Stepping through the source, I found something suspicious at [Filter.swift line 49](https://github.com/onevcat/Kingfisher/blob/master/Sources/Filter.swift#L49): in the `.data` case, the original image is decoded and returned, without touching the filter.

Hacking in `?.kf.apply(filter)` at the end of the line fixed it temporarily. I will override `process` as a more stable workaround.

Is this intentional in any way (and I'm missing a vital step somewhere), or is this a bug that can be fixed in the library?

**Comment by onevcat** [comment]
@vzsg You are right. It is a bug and I will give it a fix soon. Wait for 10 mins!

**Comment by onevcat** [comment]
Fixed in 3.1.4.
